source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -423,6 +423,11 @@ void DecodingProcessorState::clearRouteCache(const CommonResponse& common_respon
   if (!common_response.clear_route_cache()) {
     return;
   }
+  if (filter_.config().isUpstream()) {
```

Comment:
We can move this to the very beginning of the function to avoid redundant check on `common_response.clear_route_cache` in upstream filter case.

I don't feel strongly here. And we can refactor this function later when we introduce `force_clear_route_cache` config

Reply:
The order should work for now. Regarding the order with the future flag, we can think about that when that happens.